### PR TITLE
[SC-229] Added capped oracle

### PR DESCRIPTION
**Pull request by hexonaut**
Adding a capped oracle which has a maximum value.

### CODE REVIEW

File: test/CappedOracle.t.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    function setUp() public {
+        priceSource = new PriceSourceMock(0.8e8);
+        oracle      = new CappedOracle(address(priceSource), 1e8);
+    }
+
+    function test_constructor() public {
+        assertEq(oracle.latestAnswer(), 0.8e8);
+        assertEq(oracle.decimals(), 8);
+    }
```
Comment (lucas-manuel) [inline test/CappedOracle.t.sol]:
```suggestion
        assertEq(oracle.latestAnswer(), 0.8e8);
        assertEq(oracle.decimals(),     8);
```

File: test/CappedOracle.t.sol [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        assertEq(oracle.latestAnswer(), 0.8e8);
+        priceSource.setLatestAnswer(1e8);
+        assertEq(oracle.latestAnswer(), 1e8);
+    }
+
+    function test_price_above_max() public {
+        assertEq(oracle.latestAnswer(), 0.8e8);
+        priceSource.setLatestAnswer(1e8 + 1);
+        assertEq(oracle.latestAnswer(), 1e8);
+    }
```
Comment (lucas-manuel) [inline test/CappedOracle.t.sol:46]: Lets add one more prixe_below_max with `1e8 - 1`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Small changes then lgtm

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Small changes then lgtm

File: test/CappedOracle.t.sol (hunk 1 above)
Comment (hexonaut) [reply test/CappedOracle.t.sol]: done

File: test/CappedOracle.t.sol (hunk 2 above)
Comment (hexonaut) [reply test/CappedOracle.t.sol:46]: added

File: test/CappedOracle.t.sol [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    CappedOracle oracle;
+
+    function setUp() public {
+        priceSource = new PriceSourceMock(0.8e8);
+        oracle      = new CappedOracle(address(priceSource), 1e8);
+    }
+
+    function test_constructor() public {
+        assertEq(oracle.latestAnswer(), 0.8e8);
+        assertEq(oracle.decimals(),     8);
```
Comment (barrutko) [inline test/CappedOracle.t.sol:23]: I'd check the `source` address as well.

Comment (barrutko) [reply test/CappedOracle.t.sol:23]: Plus, not sure if that's necessary, but the test could also cover the revert on `_maxPrice <= 0`

**Review by barrutko — COMMENTED** [review]
Two minor comments, other than that LGTM.

File: test/CappedOracle.t.sol (hunk 3 above)
Comment (hexonaut) [reply test/CappedOracle.t.sol:23]: oops yes good catch both of these should be tested. fixed

**Review by barrutko — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**